### ISSUE TITLE
Not able to connect CRM Organization with XRM.Tools

### ISSUE THREAD

**Issue by sabby6thaugust** [davidyack/CRMSiteMapEditor#24]
Hi @davidyack

I am not to connect My CRM 2016 online organization to XRM.tool. 
I am attaching the screen shot for the error that I am getting

Please help me resolve this issue.
My email id is sabby6thaugust@outlook.com ,   sabby6thaugust@gmail.com

**Comment by ApproachICT123** [comment]
David.Harris@ApproachICT123.onmicrosoft.com

Cannot connect to XRM.too. Keeps sending me into a loop of loging in and out without being fixed. 

Your session has expired - please re-signin to continue using the application

Click here to re-authenticate

Cleared cache used different browsers, everything

**Comment by davidyack** [comment]
We hit a similar error on a trial organization recently, looks like Microsoft might have changed token validation again, and we have made a minor adjustment that has resolved it for the trial that was having the problem.  This has been deployed - can you please check if you are able to access the services now?

Thanks!

**Comment by ApproachICT123** [comment]
New problem trying to use the tools its saying. 

You currently don't have access to any CRM organizations.

**Comment by davidyack** [comment]
So that means we are able to authenticate, but the CRM discovery services I all regions are returning that you aren't licensed for any CRM organizations.   Can you confirm which region your CRM you expect to be able to connect to e.g. crm4.dynamics.com

**Comment by molyom** [comment]
Dear David, 
My CRM online 2016 is in CRM4 region and this tools doest work for me. Can you guide me how I can use it?
Thanks

**Comment by ApproachICT123** [comment]
My trial ran out.

**Comment by molyom** [comment]
Dear Sir, What do yo mean?
Can I have access to your site maper or not?

Thanks

Moly Omidvar
+34 648937222
Project Manager / IT administrator

On Mon, Aug 1, 2016 at 8:46 AM +0200, "ApproachICT123" <notifications@github.com<mailto:notifications@github.com>> wrote:

My trial ran out.

**Comment by ApproachICT123** [comment]
No. I'm not an admin or even part of the website

**Comment by davidyack** [comment]
A few answers to the questions
- CRM4 region is definitely supported
- You must be connecting to a CRM organization that has not expired and you are able to login and perform customizations on
- If no CRM organizations show on the list - check the users permissions in the office 365 portal - you must have an active subscription to CRM (trial is ok, but it can't be expired) and the user must be assigned a full license not an admin license

**Comment by molyom** [comment]
Dear David!
First of I want to appreciate you for your time and your mail.
I enjoy your course in Microsoft Academy and congratulate you for that.
Regarding to my issue, I am CRM administrator and I have full access permition to my organization.
But xrm.tool dosent find my organization and there is no way.
I have two subscription 1- professional crm 2-sandbox instance
Is it possible that I give you my pass and you check it?

Thanks a lot
Best regards

Moly Omidvar
+34 648937222
Project Manager / IT administrator

On Mon, Aug 1, 2016 at 5:36 PM +0200, "David Yack" <notifications@github.com<mailto:notifications@github.com>> wrote:

A few answers to the questions
-   CRM4 region is definitely supported
-   You must be connecting to a CRM organization that has not expired and you are able to login and perform customizations on
-   If no CRM organizations show on the list - check the users permissions in the office 365 portal - you must have an active subscription to CRM (trial is ok, but it can't be expired) and the user must be assigned a full license not an admin license

##

**Comment by molyom** [comment]
My account is XX.CRM@darSSS.ONMICROSOFT.COM (E.G.)...
But the CRM access link is darSSS.CRM4.DYNAMICS.COM (E.G.)
Your web page, just ask me for the account and how you obtain the data for CRM ADDRESS. 
 I thinks I can not work with your tools because your tools can not find my orgnization name.
(I check all of top point and there is no problem with Licence and permition, I am GLOBAL ADMIN of my organization.)

Thanks

**Comment by davidyack** [comment]
Yes, from the account, we query the discovery service for each region that determines the list of CRM organizations that show up for working with in the site. On http://xrm.tools/Home/Contact we have an email address - send me a direct email  - and we can follow up from there

Thanks
Dave
